src/game/mario_actions_moving.c

```diff
@@ -1405,13 +1405,24 @@ void common_slide_action(struct MarioState *m, u32 endAction, u32 airAction, s32
 
 s32 common_slide_action_with_jump(struct MarioState *m, u32 stopAction, u32 jumpAction, u32 airAction,
                                   s32 animation) {
+#ifdef SLOPE_FIX
+    if (m->actionTimer == 5) {
+        if (m->input & INPUT_A_PRESSED || m->actionState) {
+            return set_jumping_action(m, jumpAction, 0);
+        }
+    } else {
+        m->actionTimer++;
+        m->actionState = m->actionState ? m->input & INPUT_A_DOWN : m->input & INPUT_A_PRESSED;
+    }
+#else
```

Comment:
This isn't the most readable thing in the world, likely doesn't compile without warnings, and also potentially has functional flaws. Something like this might be better (untested though):

```suggestion
#ifdef SLOPE_FIX
    if (m->input & INPUT_A_PRESSED) {
        m->actionState = TRUE;
    } else if (!(m->input & INPUT_A_DOWN)) {
        m->actionState = FALSE;
    }

    if (m->actionTimer == 5) {
        if (m->actionState) {
            return set_jumping_action(m, jumpAction, 0);
        }
    } else {
        m->actionTimer++;
    }
#else
```

Reply:
Fair, when writing the code I was trying to optimize for speed instead of readability, though I think this code should only run once each frame, and only when mario is butt sliding anyway, so I realize readability matters more in this case. Your code works (I tested it) so I put it in because it seems much more readable than my code.